src/zenml/integrations/mlflow/steps/mlflow_deployer.py

```diff
@@ -111,98 +111,83 @@ def mlflow_model_deployer_step(
         run_name=run_name or run_name,
     )
 
+    # Fetch the model URI from the MLflow artifact store
     model_uri = ""
     if mlflow_run_id and client.list_artifacts(mlflow_run_id, model_name):
         model_uri = artifact_utils.get_artifact_uri(
             run_id=mlflow_run_id, artifact_path=model_name
         )
 
-    # fetch existing services with same pipeline name, step name and model name
+    # Fetch existing services with same pipeline name, step name and model name
     existing_services = model_deployer.find_model_server(
         pipeline_name=pipeline_name,
         pipeline_step_name=step_name,
         model_name=model_name,
     )
 
-    # create a config for the new model service
-    predictor_cfg = MLFlowDeploymentConfig(
-        model_name=model_name or "",
-        model_uri=model_uri,
-        workers=workers,
-        mlserver=mlserver,
-        pipeline_name=pipeline_name,
-        run_name=run_name,
-        pipeline_run_id=run_name,
-        pipeline_step_name=step_name,
-        timeout=timeout,
-    )
-
-    # Creating a new service with inactive state and status by default
-    service = MLFlowDeploymentService(predictor_cfg)
-    if existing_services:
-        service = cast(MLFlowDeploymentService, existing_services[0])
-
-    # check for conditions to deploy the model
-    if not model_uri:
-        # an MLflow model was not trained in the current run, so we simply reuse
-        # the currently running service created for the same model, if any
-        if not existing_services:
-            logger.warning(
-                f"An MLflow model with name `{model_name}` was not "
-                f"logged in the current pipeline run and no running MLflow "
-                f"model server was found. Please ensure that your pipeline "
-                f"includes a step with a MLflow experiment configured that "
-                "trains a model and logs it to MLflow. This could also happen "
-                "if the current pipeline run did not log an MLflow model  "
-                f"because the training step was cached."
-            )
-            # return an inactive service just because we have to return
-            # something
-            return service
+    # Check whether to deploy a new service
+    if model_uri and deploy_decision:
```

Comment:
What is the reason for this `deploy_decision`? Why one would call a deployer without a deploy decision being made?

Reply:
Main reason is that we don't have dynamic pipelines yet, so if you want to decide on the fly whether to deploy a model or not (i.e. by checking whether it outperforms previous models or similar) then you can at least pass the result of your check as an artifact into this step to achieve the behavior you want